File: relationship-manager/common/src/main/java/com/bank/rm/common/config/KafkaConfig.java
package com.bank.rm.common.config;

import com.bank.rm.common.event.KafkaTopics;
import org.apache.kafka.clients.admin.NewTopic;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.config.TopicBuilder;

@Configuration
public class KafkaConfig {

    @Bean
    public NewTopic conversationEvents() {
        return TopicBuilder.name(KafkaTopics.CONVERSATION_EVENTS).partitions(3).replicas(1).build();
    }

    @Bean
    public NewTopic profileEvents() {
        return TopicBuilder.name(KafkaTopics.PROFILE_EVENTS).partitions(3).replicas(1).build();
    }

    @Bean
    public NewTopic riskEvents() {
        return TopicBuilder.name(KafkaTopics.RISK_EVENTS).partitions(3).replicas(1).build();
    }

    @Bean
    public NewTopic recommendationEvents() {
        return TopicBuilder.name(KafkaTopics.RECOMMENDATION_EVENTS).partitions(3).replicas(1).build();
    }

    @Bean
    public NewTopic followupEvents() {
        return TopicBuilder.name(KafkaTopics.FOLLOWUP_EVENTS).partitions(3).replicas(1).build();
    }

    @Bean
    public NewTopic reminderEvents() {
        return TopicBuilder.name(KafkaTopics.REMINDER_EVENTS).partitions(3).replicas(1).build();
    }

    @Bean
    public NewTopic notificationEvents() {
        return TopicBuilder.name(KafkaTopics.NOTIFICATION_EVENTS).partitions(3).replicas(1).build();
    }

    @Bean
    public NewTopic productEvents() {
        return TopicBuilder.name(KafkaTopics.PRODUCT_EVENTS).partitions(3).replicas(1).build();
    }
}

File: relationship-manager/common/src/main/kotlin/com/bank/rm/common/config/KafkaConfig.kt
package com.bank.rm.common.config

import com.bank.rm.common.event.DomainEvent
import com.fasterxml.jackson.databind.ObjectMapper
import org.apache.kafka.clients.producer.ProducerConfig
import org.apache.kafka.common.serialization.StringSerializer
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.core.DefaultKafkaProducerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.kafka.core.ProducerFactory
import org.springframework.kafka.support.serializer.JsonSerializer

@Configuration
class KafkaProducerConfig {

    @Value("\${spring.kafka.bootstrap-servers:localhost:9092}")
    private lateinit var bootstrapServers: String

    @Bean
    fun producerFactory(objectMapper: ObjectMapper): ProducerFactory<String, DomainEvent> {
        val config = mapOf(
            ProducerConfig.BOOTSTRAP_SERVERS_CONFIG to bootstrapServers,
            ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG to StringSerializer::class.java,
            ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG to JsonSerializer::class.java,
            ProducerConfig.ACKS_CONFIG to "all",
            ProducerConfig.RETRIES_CONFIG to 3,
            ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG to true
        )
        return DefaultKafkaProducerFactory(config)
    }

    @Bean
    fun kafkaTemplate(producerFactory: ProducerFactory<String, DomainEvent>): KafkaTemplate<String, DomainEvent> =
        KafkaTemplate(producerFactory)
}
